src/main/java/duke/dependencies/parser/Parser.java

```diff
@@ -0,0 +1,223 @@
+package duke.dependencies.parser;
+
+import duke.dependencies.dukeexceptions.DukeException;
+import duke.dependencies.dukeexceptions.EmptyTaskException;
+import duke.dependencies.dukeexceptions.InvalidDateException;
+import duke.dependencies.dukeexceptions.InvalidPassException;
+import duke.dependencies.dukeexceptions.UnknownCommandException;
+import duke.dependencies.executable.Command;
+import duke.dependencies.executable.Executable;
+import duke.dependencies.storage.Storage;
+import duke.dependencies.task.Task;
+import duke.dependencies.task.TaskDate;
+
+import java.util.Locale;
+import java.util.regex.Pattern;
+
+class Parser {
+    private final Executable command;
+
+    /**
+     * Constructor for the checker object. Validates the command given/if and creates an Executable object
+     * that can be passed to the executor.
+     *
+     * @param e String to be parsed into an Executable for Executor.
+     */
+    private Parser(Executable e) {
+        this.command = e;
+    }
+
+    /**
+     * Returns the Executable object.
+     * Executable object holds the Task object inside.
+     *
+     * @return The Executable object.
+     */
+    public Executable getExecutable() {
+        return this.command;
+    }
+
+
+    /**
+     *
```

Comment:
I think a brief description for each method is needed at the start of the javadocs (before @param, @return, @throws)